### PR TITLE
Change document structure and update documents to latest

### PR DESCRIPTION
**Pull request by BewareMyPower**
The current KoP docs has a strong dependency on StreamNative website, the docs in KoP repo is outdated. This PR removes the dependency and port the docs on the website to the repository.

- Change configuration and security docs from StreamNative website to `docs/` directory
- Update some docs after multi-tenancy was supported
- Remove the outdated `DEVELOP.md` and `kafka-impl/README.md`, because they are outdated and the content is fully covered by other docs, `kafka-impl/README.md` is preserved only for maven build

The mapping from website's docs to the repo's docs are:

- https://streamnative.io/docs/latest/connect/kop/overview => some of README.md and docs/implementation.md
- https://streamnative.io/docs/latest/connect/kop/get-started => repeated with README.md
- https://streamnative.io/docs/latest/connect/kop/limitation => the 2 of 3 limitations are outdated now, the rest one is recorded in README.md 
- https://streamnative.io/docs/latest/configure/pulsar-core/kop => docs/configuration.md
- https://streamnative.io/docs/latest/secure/pulsar-core/kop => docs/security.md

### CODE REVIEW

**Comment by Huanli-Meng** [comment]
check with Jia on the platform meeting, docs update for Hub and StreamNative platform are of low priority.

File: docs/security.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+* The `username` of Kafka JAAS is the `tenant/namespace`, in which Kafka’s topics are stored in Pulsar.
+
+    For example, `public/default`.
+
+* The `password` must be your token authentication parameters from Pulsar.
+
+    For example, `token:xxx`.
+
+    The token can be created by Pulsar tokens tools. The role is the `subject` for token, it is embedded in the created token, and the broker can get `role` by parsing this token.
```
Comment (Huanli-Meng) [inline docs/security.md:18]:
```suggestion
    The token can be created by Pulsar tokens tools. The role is the `subject` for token. It is embedded in the created token, and the broker can get `role` by parsing this token.
```

File: docs/security.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+## Enable authentication on Pulsar broker
+
+To enable KoP authentication, you need to set all the options required by Pulsar token based authentication and set `saslAllowedMechanisms` (set it to`PLAIN`). The Kafka authentication is forwarded to Pulsar's JWT (Json Web Token) authentication, so you also need to configure the [JWT authentication](https://pulsar.apache.org/docs/en/security-jwt/).
```
Comment (Huanli-Meng) [inline docs/security.md:22]:
```suggestion
To enable KoP authentication, you need to set all the options required by the Pulsar token based authentication and set `saslAllowedMechanisms` (set it to`PLAIN`). The Kafka authentication is forwarded to Pulsar's JWT (Json Web Token) authentication, so you also need to configure the [JWT authentication](https://pulsar.apache.org/docs/en/security-jwt/).
```

File: docs/security.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+> **Tip**
+> For how to configure SSL keys, see [Kafka SSL](https://kafka.apache.org/documentation/#security_ssl).
+
+The following example shows how to connect KoP through SSL.
+
+1. Create SSL related keys.
+
+    This example creates related CA and jks files.
```
Comment (Huanli-Meng) [inline docs/security.md:78]:
```suggestion
    This example creates the related CA and JKS files.
```

File: docs/security.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    keytool -keystore server.truststore.jks -alias CARoot -import -file ca-cert
+    keytool -keystore client.truststore.jks -alias CARoot -import -file ca-cert
+    #Step 3
+    keytool -keystore server.keystore.jks -alias localhost -certreq -file cert-file
+    openssl x509 -req -CA ca-cert -CAkey ca-key -in cert-file -out cert-signed -days 365 -CAcreateserial -passin pass:test1234
+    keytool -keystore server.keystore.jks -alias CARoot -import -file ca-cert
+    keytool -keystore server.keystore.jks -alias localhost -import -file cert-signed
+    ```
+
+2. Configure KoP broker.
```
Comment (Huanli-Meng) [inline docs/security.md:95]:
```suggestion
2. Configure the KoP broker.
```

File: docs/security.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+    In StreamNative Platform configuration file (`${PLATFORM_HOME}/etc/pulsar/broker.conf` or `${PLATFORM_HOME}/etc/pulsar/standalone.conf`), add the related configurations that using the jks configs created in step1:
```
Comment (Huanli-Meng) [inline docs/security.md:97]:
```suggestion
    In the StreamNative Platform configuration file (`${PLATFORM_HOME}/etc/pulsar/broker.conf` or `${PLATFORM_HOME}/etc/pulsar/standalone.conf`), add the related configurations that using the jks configurations created in Step 1:
```

File: docs/security.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    listeners=PLAINTEXT://localhost:9092,SSL://localhost:9093
+
+    kopSslKeystoreLocation=/Users/kop/server.keystore.jks
+    kopSslKeystorePassword=test1234
+    kopSslKeyPassword=test1234
+    kopSslTruststoreLocation=/Users/kop/server.truststore.jks
+    kopSslTruststorePassword=test1234
+    ```
+
+3. Configure Kafka client.
```
Comment (Huanli-Meng) [inline docs/security.md:109]:
```suggestion
3. Configure the Kafka client.
```

File: docs/security.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+    (1) Prepare a file named `client-ssl.properties` containing the following information.
```
Comment (Huanli-Meng) [inline docs/security.md:111]:
```suggestion
    (1) Prepare a file named `client-ssl.properties`. The file contains the following information.
```

File: docs/security.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+    ```shell
+    security.protocol=SSL
+    ssl.truststore.location=client.truststore.jks
+    ssl.truststore.password=test1234
+    ssl.endpoint.identification.algorithm=
+    ```
+
+    (2) Verify console-producer and console-consumer.
```
Comment (Huanli-Meng) [inline docs/security.md:120]:
```suggestion
    (2) Verify the console-producer and the console-consumer.
```

File: integrations/README.md [hunk 9]
```diff
@@ -2,6 +2,20 @@
 
 # Produce and Consume support
 
+## Java
+
+Kafka official clients, the supported versions are:
```
Comment (Huanli-Meng) [inline integrations/README.md:7]:
```suggestion
The following official Kafka clients are available:
```

File: kafka-impl/README.md [hunk 10]
```diff
@@ -2,187 +2,4 @@
...
-String pasword = "token:xxx";
-
-String jaasTemplate = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"%s\" password=\"%s\";";
-String jaasCfg = String.format(jaasTemplate, tenant, password);
-props.put("sasl.jaas.config", jaasCfg);
-props.put("security.protocol", "SASL_PLAINTEXT");
-props.put("sasl.mechanism", "PLAIN");
-
-```
+See [README.md](https://github.com/streamnative/kop/blob/master/README.md) for detail.
```
Comment (Huanli-Meng) [inline kafka-impl/README.md:5]:
```suggestion
For details, see [README](https://github.com/streamnative/kop/blob/master/README.md) for detail.
```

**Review by Huanli-Meng — COMMENTED** [review]
Leave my comments.

File: README.md [hunk 11]
```diff
@@ -64,18 +67,18 @@ add configurations in Pulsar's configuration file, such as `broker.conf` or `sta
...
+    | Property | Set it to the following value | Default value |
+    | :------- | :---------------------------- | :------------ |
+    | `messagingProtocols` | kafka | null |
+    | `protocolHandlerDirectory`| Location of KoP NAR file | ./protocols |
+    | `allowAutoTopicCreationType`| partitioned | non-partitioned |
+
     **Example**
 
-    ```
+    ```properties
```
[the remark this answers is not in the document]
Comment (BewareMyPower) [reply README.md:81]: The Pulsar config files that have the suffix `*.conf` are Java properties files. The file type is used for syntax highlight in markdown. See [install-kop-protocol-handler](https://github.com/BewareMyPower/kop/tree/bewaremypower/docs-update#install-kop-protocol-handler) for the render effect.

**Comment by BewareMyPower** [comment]
The changes have been made, PTAL, @Huanli-Meng

**Review by Huanli-Meng — APPROVED** [review]
LGTM

**Pull request merged**